cdc/sink/codec/avro_test.go

```diff
@@ -58,43 +59,98 @@ func (s *avroBatchEncoderSuite) TearDownSuite(c *check.C) {
+	types.SetTypeFlag(&ft.Flag, flag, true)
+	return ft
+}
+
+func setElems(ft *types.FieldType, elems []string) *types.FieldType {
+	ft.Elems = elems
+	return ft
+}
+
 func (s *avroBatchEncoderSuite) TestAvroEncodeOnly(c *check.C) {
 	defer testleak.AfterTest(c)()
-	avroCodec, err := goavro.NewCodec(`
-        {
-          "type": "record",
-          "name": "test1",
-          "fields" : [
-            {"name": "id", "type": ["null", "int"], "default": null},
-			{"name": "myint", "type": ["null", "int"], "default": null},
-			{"name": "mybool", "type": ["null", "int"], "default": null},
-			{"name": "myfloat", "type": ["null", "float"], "default": null},
-			{"name": "mybytes", "type": ["null", "bytes"], "default": null},
-			{"name": "ts", "type": ["null", {"type": "long", "logicalType": "timestamp-millis"}], "default": null}
-          ]
-        }`)
-
-	c.Assert(err, check.IsNil)
 
 	table := model.TableName{
 		Schema: "testdb",
-		Table:  "test1",
+		Table:  "TestAvroEncodeOnly",
 	}
 
-	r, err := avroEncode(&table, s.encoder.valueSchemaManager, 1, []*model.Column{
+	cols := []*model.Column{
 		{Name: "id", Value: int64(1), Type: mysql.TypeLong},
 		{Name: "myint", Value: int64(2), Type: mysql.TypeLong},
 		{Name: "mybool", Value: int64(1), Type: mysql.TypeTiny},
 		{Name: "myfloat", Value: float64(3.14), Type: mysql.TypeFloat},
-		{Name: "mybytes", Value: []byte("Hello World"), Type: mysql.TypeBlob},
+		{Name: "mybytes1", Value: []byte("Hello World"), Flag: model.BinaryFlag, Type: mysql.TypeBlob},
+		{Name: "mybytes2", Value: []byte("Hello World"), Flag: model.BinaryFlag, Type: mysql.TypeMediumBlob},
+		{Name: "mybytes3", Value: []byte("Hello World"), Flag: model.BinaryFlag, Type: mysql.TypeTinyBlob},
+		{Name: "mybytes4", Value: []byte("Hello World"), Flag: model.BinaryFlag, Type: mysql.TypeLongBlob},
+		{Name: "mybytes5", Value: []byte("Hello World"), Flag: model.BinaryFlag, Type: mysql.TypeVarString},
+		{Name: "mybytes6", Value: []byte("Hello World"), Flag: model.BinaryFlag, Type: mysql.TypeString},
+		{Name: "mybytes7", Value: []byte("Hello World"), Flag: model.BinaryFlag, Type: mysql.TypeVarchar},
+		{Name: "mystring1", Value: "Hello World", Type: mysql.TypeBlob},
+		{Name: "mystring2", Value: "Hello World", Type: mysql.TypeMediumBlob},
+		{Name: "mystring3", Value: "Hello World", Type: mysql.TypeTinyBlob},
+		{Name: "mystring4", Value: "Hello World", Type: mysql.TypeLongBlob},
+		{Name: "mystring5", Value: "Hello World", Type: mysql.TypeVarString},
+		{Name: "mystring6", Value: "Hello World", Type: mysql.TypeString},
+		{Name: "mystring7", Value: "Hello World", Type: mysql.TypeVarchar},
+		{Name: "myenum", Value: uint64(1), Type: mysql.TypeEnum},
+		{Name: "myset", Value: uint64(1), Type: mysql.TypeSet},
 		{Name: "ts", Value: time.Now().Format(types.TimeFSPFormat), Type: mysql.TypeTimestamp},
 		{Name: "myjson", Value: "{\"foo\": \"bar\"}", Type: mysql.TypeJSON},
-	}, time.Local)
+	}
+
+	colInfos := []rowcodec.ColInfo{
```

Comment:
Maybe we can organize this data by structured arrays? Controlling correspondence by index may be difficult with large test data.

Reply:
Yes. Actually using arrays restricts the usage pattern. We have to iterate the array(currently no problem). I tried to use the tp map, but it can't be used since the column `Id` is lost in `Column` struct. Add fields to `Column` will cause much code churn. Any suggestions?